lib/puppet/functions/mysql/password.rb

```diff
@@ -22,7 +22,7 @@
   def password(password, sensitive = false)
     password = password.unwrap if password.is_a?(Puppet::Pops::Types::PSensitiveType::Sensitive)
 
-    result_string = if %r{\*[A-F0-9]{40}$}.match?(password)
+    result_string = if %r{^\*[A-F0-9]{40}$}.match?(password)
```

Comment:
I just have a few minutes right now, but I'm wondering why we have this regex at all? Does it check if it's a hash and otherwise runs into the last else block and runs SHA1?

Reply:
I think the intention is if it's already a Hash, then don't hash it, but since the regex isn't anchored it _can_ pick up other passwords.  The other password looks incredibly unlikely in the real world though.  But I guess @Ramesh7 bumped into this in their environment??

Reply:
@bastelfreak yeah, if its hash then it will return that otherwise it will hash it with SHA1.
@alexjfisher yeah, recently we encountered this and found there is flow in regex which needs to fix, so took it here.